Cql/Cql.Packaging/ResourceWriters/FhirResourceWriter.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Hl7.Fhir.Model;
+using Hl7.Fhir.Serialization;
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+
+namespace Hl7.Cql.Packaging.ResourceWriters
+{
+    /// <summary>
+    /// Writes resources as FHIR files
+    /// </summary>
+    public class FhirResourceWriter : ResourceWriter
+    {
+        /// <summary>
+        /// Instantiates a new resource writer
+        /// </summary>
+        /// <param name="outDirectory">the output directory</param>
+        /// <param name="logger">logger</param>
+        public FhirResourceWriter(DirectoryInfo outDirectory, ILogger logger) : base(outDirectory, logger)
+        {
+        }
+
+        /// <summary>
+        /// Writes the specified resources to the output directory in the FHIR format
+        /// </summary>
+        /// <param name="resources">the resources to write</param>
+        public override void WriteResources(IEnumerable<Resource> resources)
+        {
+            var options = new JsonSerializerOptions()
+                .ForFhir(typeof(Resource).Assembly)
```

Comment:
This is incorrect, see https://docs.fire.ly/projects/Firely-NET-SDK/en/latest/parsing/system-text-json-serialization.html.

This is the way to do it:

```
Patient p = new() {  };
var options = new JsonSerializerOptions().ForFhir(ModelInfo.ModelInspector).Pretty();
string patientJson = JsonSerializer.Serialize(p, options);
```

Reply:
Also, the Json.Text framework uses the options object to cache the reflection metadata, so it's better to keep it around if you do multiple `WriteResources()`.

Reply:
Updated.